analyzer/analysis/requirementsAnalysis.py

```diff
@@ -84,21 +85,23 @@ def get_control(c: TransmissionComponent) -> ControlType:
 
         l.info(f"{v}:     {load_anno}    {uncontrolled_anno}")
 
-        if uncontrolled_anno == None and (load_anno == None  or load_anno.controlled == True):
+        if uncontrolled_anno == None and (load_anno == None or load_anno.controlled == True):
             controlled = True
         else:
             has_uncontrolled_components = True
 
-    if not controlled:  # and has_uncontrolled_components:
-        # return ControlType.REQUIRES_MEM_MASSAGING
-        return ControlType.NO_CONTROL
-
-    if controlled and not has_uncontrolled_components:
-        return ControlType.CONTROLLED
-
-    if controlled and has_uncontrolled_components:
-        return ControlType.REQUIRES_MEM_LEAK
+    if controlled:
+        if has_uncontrolled_components:
+            return ControlType.REQUIRES_MEM_LEAK
+        else:
+            return ControlType.CONTROLLED
+    else:
+        if has_uncontrolled_components and report_massaging:
```

Comment:
Can `has_uncontrolled_components` at this point ever by `false`? It seems not, as every symbolic expr has at least 1 var. And every expr with a mem/reg requirement is symbolic.

Reply:
true, will fix. we might want to rewrite this anyways, it's quite confusing I think.

Reply:
ok please check latest force-pushed version, if it makes sense then we're ready to go